### PR TITLE
TIM1 configuration

### PR DESCRIPTION
**Pull request by cua-cua**
The timer 1 blinks a led each second.
It is configured to generate an interruption when the update event occurs. 
The interruption executes a function where the led is toggling. 
This pull request should be reviewed after #56 one.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -20,11 +20,18 @@ int _write(int file, char *ptr, int len);
  * This output frequency is possible thanks to the Phase Locked Loop (PLL)
  * multiplier.
  *
+ * The default values for preescalers are:
+ * ADC prescaler(Max. 14MHz): Divider 8 / Set 9MHz
+ * APB1 prescaler(Max. 36MHz): Divider 2 / Set 36MHz
+ * APB2 prescaler(Max. 72MHz): Divider 1 / Set 72 MHz
+ * AHB prescaler(Max. 72MHz): Divider 1 / Set 72 MHz
+
```
Comment (Peque) [inline src/main.c]: Not sure this should go here. Maye a "@see" note to point to the reference manual section? Or maybe better in Athena project?

Anyway if it is a list, I would format it like:

```c
/**
 * [...]
 * 
 * The default values for the prescalers are:
 *
 * - ADC prescaler (max. 14 MHz): divider 8 / set 9 MHz.
 * - ...
```

Note the space before parenthesis, new line before list, no caps if no dot/stop before the word and a space before units (I think the only exception is for degrees).

What does "divider X / set Y" means here? Maybe it is not clear. :sweat_smile:

File: src/main.c [hunk 2]
```diff
@@ -334,6 +350,55 @@ static void setup_adc(void)
 
 
 /**
+ * @brief TIM1 Interruption routine
+ *
+ *- Update event interrupt flag management
+ *- Led toggling each second
```
Comment (Peque) [inline src/main.c]: Stops/dots after the sentences? :stuck_out_tongue_winking_eye: 

I think you are missing a space between `*` and the `-` in both list items. :blush:

Comment (Peque) [reply src/main.c]: LED is an achronym.

File: src/main.c [hunk 3]
```diff
@@ -70,6 +81,11 @@ static void setup_gpio(void)
 	/* ADC sensors */
 	gpio_set_mode(GPIOA, GPIO_MODE_INPUT, GPIO_CNF_INPUT_ANALOG,
 		GPIO4 | GPIO5 | GPIO6 | GPIO7);
+
+	/* Led test */
```
Comment (Peque) [inline src/main.c]: LED. :wink:

File: src/main.c [hunk 4]
```diff
@@ -334,6 +350,55 @@ static void setup_adc(void)
...
+		gpio_toggle(GPIOA, GPIO7);
+	}
+}
+
+
+/**
+ * @brief Timer 1 (TIM1) setup.
+ *
+ * The timer 1 generates an update event interruption that invokes the
+ * function tim1_up_isr.
```
Comment (Peque) [inline src/main.c]: Missing one blank line before starting the list.

File: src/main.c [hunk 5]
```diff
@@ -334,6 +350,55 @@ static void setup_adc(void)
...
+ * - Enable TIM1 interruption due an update event on the system.
+ * - Set timer 1 default values.
+ * - Configure the base time (No clock division ratio - No aligned mode(edge)
+ *   direction up).
```
Comment (Peque) [inline src/main.c]: I think it should be "no" (no caps). Missing space before `(` in `(edge)`.

File: src/main.c [hunk 6]
```diff
@@ -334,6 +350,55 @@ static void setup_adc(void)
...
+	}
+}
+
+
+/**
+ * @brief Timer 1 (TIM1) setup.
+ *
+ * The timer 1 generates an update event interruption that invokes the
+ * function tim1_up_isr.
+ * - Enable TIM1 interruption due an update event on the system.
```
Comment (Peque) [inline src/main.c]: What do you mean? I think the sentence might be incorrect (in English I mean).

File: src/main.c [hunk 7]
```diff
@@ -43,6 +50,10 @@ static void setup_clock(void)
 
 	/* ADC */
 	rcc_periph_clock_enable(RCC_ADC1);
+
+	/* TIM1 */
+	rcc_periph_clock_enable(RCC_TIM1);
+
```
Comment (Peque) [inline src/main.c]: I would avoid this newline before the closing `}` just for consistency (not used in any other function).

File: src/main.c (hunk 1 above)
Comment (cua-cua) [reply src/main.c]: Included a @see section with the reference manual, and comment rewritten. Without issues I hope :)

File: src/main.c (hunk 7 above)
Comment (cua-cua) [reply src/main.c]: fixed with clang-format ^^

File: src/main.c (hunk 3 above)
Comment (cua-cua) [reply src/main.c]: Forr me is a word, fixed anyway

File: src/main.c (hunk 2 above)
Comment (cua-cua) [reply src/main.c]: ok

File: src/main.c (hunk 4 above)
Comment (cua-cua) [reply src/main.c]: done

File: src/main.c (hunk 6 above)
Comment (cua-cua) [reply src/main.c]: you are right, I rewrote the sentence to be clear "Enable an interruption of type timer 1 update event on the system"

File: src/main.c (hunk 5 above)
Comment (cua-cua) [reply src/main.c]: ok, and edge deleted

**Pull request merged**